celltype_annotation/analysis/cell-assign-sarcoma.Rmd

```diff
@@ -0,0 +1,481 @@
+```
+
+
+```{r}
+# create a binary matrix of genes by cell type markers 
+cellmarker_mtx <- muscle_markers_df |>
+  dplyr::select(cellontology_id, Symbol) |> 
+  tidyr::pivot_wider(id_cols = Symbol,
+                     names_from = cellontology_id,
+                     values_from = cellontology_id,
+                     values_fn = length,
+                     values_fill = 0) |> 
+  tibble::column_to_rownames("Symbol") |>
+  dplyr::mutate(other = 0)
+
+# replace length with 1
+cellmarker_mtx[cellmarker_mtx > 1] <- 1
+```
+
+
+
+```{r}
+# convert rownames to ensembl ids
+rowdata_df <- rowData(processed_sce) |>
+  as.data.frame() |>
+  tibble::rownames_to_column("ensembl_id") |>
+  dplyr::select(ensembl_id, gene_symbol)
+
+# merge reference with rowdata and replace gene symbols with ensembl
+cellmarker_mtx <- cellmarker_mtx |>
+  tibble::rownames_to_column("gene_symbol") |>
+  dplyr::left_join(rowdata_df) |>
+  # drop any rows where there is no ensembl id
+  tidyr::drop_na(ensembl_id) |>
+  dplyr::select(-gene_symbol) |>
+  dplyr::relocate(ensembl_id)
+```
+
+```{r}
+# export mtx 
+readr::write_csv(cellmarker_mtx, cell_marker_mtx_file)
+```
+
+```{r,  eval = FALSE}
+# run cell assign with muscle ref
+cellassign_call <- glue::glue("
+python ../scripts/cell-assign.py \
+  --input_anndata '{anndata_file}' \
+  --output_predictions '{cellmarker_prediction_file}' \
+  --reference '{cell_marker_mtx_file}'
+")
+
+system(cellassign_call)
+```
+
+
+```{r}
+# read in prediction scores 
+cellmarker_predictions <- readr::read_tsv(cellmarker_prediction_file)
+```
+
+```{r}
+# grab cell label names corresponding to cell ontology ids
+cl_ont <- ontoProc::getCellOnto()
+
+# replace column names(ontology ids) with cell label
+cellmarker_predictions <- cellmarker_predictions |>
+  dplyr::rename_with(
+    \(cl_id) 
+    cl_ont$name[stringr::str_replace(cl_id, "_", ":")],
+    !any_of(c("barcode", "other"))
+)
+```
+
+
+
+```{r}
+# get assignments for each cell from prediction matrix 
+cellmarker_assignments <- get_celltype_assignments(cellmarker_predictions)
+
+# print table of assignments
+cellmarker_assignments |>
+  dplyr::count(celltype) |>
+  dplyr::arrange(desc(n))
+  
+```
+
+```{r}
+# heatmap of prediction scores 
+cellmarker_predictions |>
+  dplyr::select(-barcode) |> 
+  as.matrix() |> 
+  pheatmap::pheatmap(show_rownames = FALSE)
+```
+
+Here most of the cells appear to be assigned to macrophages... 
+According to the known cell type annotations most of the cell types here are tumor cells and of those tumor cells they are mostly myoblasts. 
+
+```{r}
+# print out markers that were associated with macrophages during assignment
+muscle_markers_df |>
+  dplyr::filter(cell_name == "Macrophage")
+```
+
+What's strange is that CD163 and CD68 aren't present in our SCE object, only PTPRC and it is not highly expressed.
+
+```{r}
+processed_sce$cellmarker_assignments <- cellmarker_assignments$celltype
```

Comment:
Worth checking that the barcodes are in the same order here before adding the annotations to the SCE object